### PR TITLE
feat(command): support HSCAN command

### PR DESCRIPTION
**Pull request by ruojieranyishen**
## Summary by CodeRabbit

* **New Features**
  * HSCAN: resumable, cursor-based scanning of hash fields with MATCH and COUNT; returns next cursor and field/value pairs as a READONLY hash command.
  * Persistent scan state: next-scan positions are stored to support paginated/resumable iterations.

* **Improvements**
  * Improved pattern handling with optimized tail-wildcard detection for faster pattern scans.

* **Tests**
  * Added comprehensive HSCAN tests for pagination, pattern filtering, edge cases, and argument/metadata validation.

### CODE REVIEW

File: src/storage/src/redis_hashes.rs [hunk 1]
```diff
@@ -1083,4 +1084,200 @@ impl Redis {
...
+                }
+
+                Ok((next_cursor, result_fields))
+            }
+            None => {
+                // Key not found, return empty result with cursor 0
+                Ok((0, Vec::new()))
+            }
+        }
+    }
```
Comment (ruojieranyishen) [inline src/storage/src/redis_hashes.rs:1247]: reslove

**Comment by ruojieranyishen** [comment]
@AlexStocks please review my PR, thinks.

**Review by AlexStocks — APPROVED** [review]
